src/klein/_decorators.py

```diff
@@ -43,9 +52,10 @@ def modified(modification, original, modifier=None):
         likely calls it.
     """
     def decorator(wrapper):
-        result = (named(modification + ' for ' + original.__name__)
-                  (wraps(original)(wrapper)))
-        result.__original__ = original
+        # type: (C) -> C
+        namer = named(modification + ' for ' + original.__name__)
+        result = cast(C, namer(wraps(original)(wrapper)))
```

Comment:
`cast` needed because `wraps` doesn't appear to return the same `Callable` signature as its input.
I suspect that's a bug; if it is and it's fixed, `mypy` will flag this as redundant.